### ISSUE TITLE
Feature-Request: Collapse Functions by Default in Outline-View

### ISSUE THREAD

**Issue by wottpal** [facebook-atom/atom-ide-ui#166]
Hey,
it would be nice to have an option that by default all functions are collapsed in the outline view. Another way I could imagine is collapsing all functions if otherwise the outline-views height is larger than the viewport-height (outline-view is scrollable).

<img width="300" alt="bildschirmfoto 2018-02-13 um 23 22 20" src="https://user-images.githubusercontent.com/3930150/36177040-c2d0e914-1114-11e8-9ea0-bb88ab795419.png">

VS

<img width="300" alt="bildschirmfoto 2018-02-13 um 23 22 04" src="https://user-images.githubusercontent.com/3930150/36177042-c2f286dc-1114-11e8-9be7-8f8404cbce6b.png">

Dennis

**Comment by wottpal** [comment]
Bonus: A counter on the right of a function could be added to indicate how many items are underneath but maybe another issue should be opened for that one.

**Comment by matthewwithanm** [comment]
Thanks for the request! @wbinnssmith whatcha think?

**Comment by wbinnssmith** [comment]
overall I like the idea.

There are a few cases where collapsing things might be less useful. For example, if there's a single top-level function with a number of useful locals defined, collapsing would leave us with a single visible node in the tree with a bunch of useful stuff underneath it.

**Comment by damieng** [comment]
I think though there won't be any kind of heuristic that will be globally useful across languages.  What we might consider is adding a default 'is expanded' property on each node in the outline and let individual packages determine the right level of initial expansion.

**Comment by damieng** [comment]
One option here might be to have a drop-down next to the filter box that allows people to check on/off the kinds they want displayed in the list. I know somebody people don't even want to see variables which might eliminate the need for functions to be collapsed at all.

**Comment by LeaVerou** [comment]
Yes please. I almost *never* want to navigate by variable definitions and it's just wasting vertical space. I'd vote for having them always collapsed by default.
